packages/compass-connections-navigation/src/tree-data.ts

```diff
@@ -0,0 +1,288 @@
+    return sidebarData;
+  }
+
+  const areCollectionsReady = ['ready', 'refreshing', 'error'].includes(
+    collectionsStatus
+  );
+  const placeholdersLength = Math.min(
+    collectionsLength,
+    MAX_COLLECTION_PLACEHOLDER_ITEMS
+  );
+
+  if (!areCollectionsReady) {
+    return sidebarData.concat(
+      Array.from({ length: placeholdersLength }, (_, index) => ({
+        level: level + 1,
+        type: 'placeholder' as const,
+        colorCode,
+        maxNestingLevel,
+        id: `${connectionId}.${id}.placeholder.${index}`,
+      }))
+    );
+  }
+
+  return sidebarData.concat(
+    collections.map(({ _id: id, name, type }) => ({
+      id: `${connectionId}.${id}`, // id is the namespace of the collection, so includes db as well
+      level: level + 1,
+      name,
+      type: type as 'collection' | 'view' | 'timeseries',
+      setSize: 0,
+      posInSet: 0,
+      colorCode,
+      connectionId,
+      namespace: id,
+      maxNestingLevel,
+    }))
+  );
+};
+
+export function getMaxNestingLevel(isSingleConnection: boolean): number {
+  return isSingleConnection ? 2 : 3;
+}
+
+/**
+ * Converts a list connections to virtual tree items.
+ *
+ * When isSingleConnection is true, the connections are treated as a single connection mode
+ * and only two levels of items are shown: databases and collections.
+ *
+ * The IDs of the items are just to be used by the tree to correctly identify the items and
+ * do not represent the actual IDs of the items.
+ *
+ * @param connections - The connections.
+ * @param isSingleConnection - Whether the connections are a single connection.
+ * @param expandedItems - The expanded items.
+ */
+export function getVirtualTreeItems(
+  connections: Connection[],
+  isSingleConnection: boolean,
+  expandedItems: Record<string, false | Record<string, boolean>> = {}
+): SidebarTreeItem[] {
+  let treeList: SidebarTreeItem[] = [];
+  if (!isSingleConnection) {
+    treeList = connections.flatMap((connection) =>
+      connectionToItems({
+        connection,
+        expandedItems,
+        maxNestingLevel: getMaxNestingLevel(isSingleConnection),
+      })
+    );
+  } else {
+    const connection = connections[0];
+    const dbExpandedItems = expandedItems[connection.connectionInfo.id] || {};
+    treeList = connection.databases.flatMap((database) => {
+      return databaseToItems({
+        connectionId: connection.connectionInfo.id,
+        database,
+        expandedItems: dbExpandedItems,
+        level: 1,
+        maxNestingLevel: getMaxNestingLevel(isSingleConnection),
+      });
+    });
+  }
+
+  // While generating a flat list of tree items, its is not possible to know the exact number
+  // of items in the list. So we will generate the list first and then map it to have correct setSize and posInSet.
+  const setSize = treeList.length;
+  return treeList.map((item, index) => {
+    const posInSet = index + 1;
+    return {
+      ...item,
+      setSize,
+      posInSet,
+    };
+  });
```

Comment:
I don't think this is correct. The setsize and posinset should be set relative to the tree level, not for the whole tree, this is what aria spec expects, it's not just a total number of items in the list

Reply:
fixed it in [a082a00](https://github.com/mongodb-js/compass/pull/5837/commits/a082a003b94ee45ab865f60e74ef0066311a7169)

Reply:
nice catch :)